subsys/net/lib/shell/iface.c

```diff
@@ -536,10 +541,19 @@ static int cmd_net_set_mac(const struct shell *sh, size_t argc, char *argv[])
 		goto err;
 	}
 
-	if ((net_bytes_from_str(mac_addr, sizeof(params.mac_address), argv[2]) < 0) ||
-	    !net_eth_is_addr_valid(&params.mac_address)) {
-		PR_WARNING("Invalid MAC address: %s\n", argv[2]);
-		goto err;
+	if (!strncasecmp(argv[2], "random", 6)) {
+		char random_mac_addr[WIFI_MAC_ADDR_LEN];
+
+		sys_rand_get(random_mac_addr, WIFI_MAC_ADDR_LEN);
+		random_mac_addr[0] = (random_mac_addr[0] & UNICAST_MASK) | LOCAL_BIT;
+
+		strncpy(mac_addr, random_mac_addr, WIFI_MAC_ADDR_LEN);
```

Comment:
Why not use the buffer directly?

```suggestion
		sys_rand_get(mac_addr, NET_ETH_ADDR_LEN);
		mac_addr[0] = (mac_addr[0] & UNICAST_MASK) | LOCAL_BIT;
```